packages/firebase_vertexai/firebase_vertexai/lib/src/live_session.dart

```diff
@@ -0,0 +1,160 @@
+  late StreamSubscription _wsSubscription;
+
+  /// Sends content to the server.
+  ///
+  /// [input] (optional): The content to send.
+  /// [turnComplete] (optional): Indicates if the turn is complete. Defaults to false.
+  Future<void> send({
+    Content? input,
+    bool turnComplete = false,
+  }) async {
+    _checkWsStatus();
+    var clientMessage = input != null
+        ? LiveClientContent(turns: [input], turnComplete: turnComplete)
+        : LiveClientContent(turnComplete: turnComplete);
+    var clientJson = jsonEncode(clientMessage.toJson());
+
+    _ws.sink.add(clientJson);
+  }
+
+  /// Sends realtime input (media chunks) to the server.
+  ///
+  /// [mediaChunks]: The list of media chunks to send.
+  Future<void> sendMediaChunks({
+    required List<InlineDataPart> mediaChunks,
+  }) async {
+    _checkWsStatus();
+    var clientMessage = LiveClientRealtimeInput(mediaChunks: mediaChunks);
+    var clientJson = jsonEncode(clientMessage.toJson());
+    _ws.sink.add(clientJson);
+  }
+
+  /// Starts streaming media chunks to the server from the provided [mediaChunkStream].
+  ///
+  /// This function asynchronously processes each [InlineDataPart] from the given
+  /// [mediaChunkStream] and sends it to the server via the WebSocket connection.
+  ///
+  /// Parameters:
+  /// - [mediaChunkStream]: The stream of [InlineDataPart] objects to send to the server.
+  Future<void> startMediaStream(Stream<InlineDataPart> mediaChunkStream) async {
+    _checkWsStatus();
+
+    try {
+      await for (final chunk in mediaChunkStream) {
+        await _sendMediaChunk(chunk);
+      }
+    } catch (e) {
+      throw VertexAISdkException(e.toString());
+    } finally {
+      log('Stream processing completed.');
+    }
+  }
+
+  Future<void> _sendMediaChunk(InlineDataPart chunk) async {
+    var clientMessage = LiveClientRealtimeInput(
+        mediaChunks: [chunk]); // Create a list with the single chunk
+    var clientJson = jsonEncode(clientMessage.toJson());
+    _ws.sink.add(clientJson);
+  }
+
+  /// Receives messages from the server.
+  ///
+  /// Returns a [Stream] of [LiveServerMessage] objects representing the
+  /// messages received from the server. The stream will stops once the server
+  /// sends turn complete message.
+  Stream<LiveServerMessage> receive() async* {
+    _checkWsStatus();
+
+    await for (final result in _messageController.stream) {
+      yield result;
+      if (result is LiveServerContent &&
+          result.turnComplete != null &&
+          result.turnComplete!) {
+        break; // Exit the loop when the turn is complete
+      }
+    }
+  }
+
+  /// Receives messages from the server and invokes the [callback] function with each message.
+  ///
+  /// This function asynchronously processes messages from the server and passes each
+  /// [LiveServerMessage] to the provided [callback] function. The operation
+  /// will stops once the server sends turn complete message.
+  Future<void> receiveWithCallback(
```

Comment:
Are we adding this one solely because it's added in the other languages? The `Stream<LiveServerMessage>` is this idiomatic Dart approach and I think it would be better to have only one API if we can.

If we omit this API users only need to do `await session.receive().forEach(callback);` instead of `await session.receiveWithCallback(callback)`.